cmd/frontend/graphqlbackend/search.go

```
@@ -587,19 +587,7 @@ func filterRepoHasCommitAfter(ctx context.Context, revisions []*search.Repositor
 			var specifiers []search.RevisionSpecifier
 			for _, rev := range revs.Revs {
 				ok, err := git.HasCommitAfter(ctx, revs.GitserverRepo(), after, rev.RevSpec)
-				if err != nil {
-					if gitserver.IsRevisionNotFound(err) {
-						continue
-					}
-
-					if ctx.Err() != nil {
-						err = errors.New("timeout due to repohascommitafter: filter, please try setting a larger timeout: in your query (we are improving this, see https://github.com/sourcegraph/sourcegraph/issues/4614)")
-					}
-
-					run.Error(err)
-					continue
-				}
-				if ok {
+				if ok && err != nil {
```

It seems the error will be lost, even if it's not important in most cases. What about print a trace log?

Alternatively, should we go detecting specific errors and do `continue` like the `if gitserver.IsRevisionNotFound(err) {` we have right now.

cc @slimsag since he made the change about timeout error.

---

Got it. I'll switch to adding a case for repo not found.